libs/hooks/useUpkeeps/index.tsx

```diff
@@ -0,0 +1,122 @@
+import { poolList, ONE_HOUR } from '@libs/constants/poolLists';
+import { StaticPoolInfo } from '@libs/types/General';
+import { ARBITRUM } from '@libs/constants';
+import { calcSkew, calcTokenPrice } from '@tracer-protocol/tracer-pools-utils';
+
+export type Upkeep = {
+    pool: string;
+    timestamp: number;
+    newPrice: number;
+    oldPrice: number;
+    tvl: number;
+    longTokenBalance: number;
+    longTokenSupply: number;
+    shortTokenBalance: number;
+    shortTokenSupply: number;
+    longTokenPrice: number;
+    shortTokenPrice: number;
+    skew: number;
+};
+
+type RawUpkeep = {
+    pool_address: string;
+    long_token_supply: string;
+    short_token_supply: string;
+    long_balance: string;
+    short_balance: string;
+    old_price: string;
+    new_price: string;
+    network: string;
+    block_timestamp: string;
+};
+
+const POOLS_API = process.env.NEXT_PUBLIC_POOLS_API;
+
+// const useUpkeeps
+export const useUpkeeps: (network: AvailableNetwork | undefined) => Record<string, Upkeep[]> = (network) => {
+    const [upkeeps, setUpkeeps] = useState<Record<string, Upkeep[]>>({});
+
+    useEffect(() => {
+        let mounted = true;
+        const poolInfo: StaticPoolInfo = poolList[(network ?? ARBITRUM) as AvailableNetwork][0];
+        const USDC_DECIMALS = poolInfo?.quoteToken?.decimals ?? 18;
+        const fetchUpkeeps = async () => {
+            const now = Math.floor(Date.now() / 1000);
+            const from = now - (poolInfo?.updateInterval || ONE_HOUR).times(2).toNumber();
+
+            const rawUpkeeps = await fetch(`${POOLS_API}/upkeeps?network=${network}&from=${from}`)
+                .then((res) => {
+                    return res.json();
+                })
+                .catch((error) => {
+                    console.error('Failed to fetch upkeeps', error);
+                    return [];
+                });
+
+            if (rawUpkeeps.message) {
+                console.info('Fetched upkeeps', rawUpkeeps.message);
+            } else if (rawUpkeeps.data) {
+                if (mounted) {
+                    const upkeepMapping: Record<string, Upkeep[]> = {};
+
+                    for (const upkeep of rawUpkeeps.data) {
+                        if (upkeepMapping[upkeep.pool_address]) {
+                            upkeepMapping[upkeep.pool_address].push(parseUpkeep(upkeep, USDC_DECIMALS));
+                        } else {
+                            upkeepMapping[upkeep.pool_address] = [parseUpkeep(upkeep, USDC_DECIMALS)];
+                        }
+                    }
+
+                    for (const pool of Object.keys(upkeepMapping)) {
+                        // array is sorted by newest to oldest such that we can trim the last 2 upkeeps
+                        upkeepMapping[pool] = upkeepMapping[pool].sort((a, b) => b.timestamp - a.timestamp);
+                    }
+                    setUpkeeps(upkeepMapping);
+                }
+            }
```

Comment:
although there is no message included with the data currently, we should not have this assumption that the presence of message means there is no data

Reply:
Ill put in a TODO??